Sources/GRPCHTTP2TransportNIOPosix/TLSConfig.swift

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+public struct TLSConfig: Sendable {
```

Comment:
I think some of the things we discussed offline got lost. This should be a general purpose `TLSConfig` namespace (`enum`) for the posix transports but shouldn't be a config.

The posix server and client configs should have their own `TLS` configs made up of components from this namespace; they are separate and different configs. As an example the client shouldn't have a `requireALPN`, and the default config for a client is different to the default config for a server (because no certs/private key are required for the client).

So we should be:

```swift
public enum TLSConfig {
  public struct SerializationFormat { ... }
  public struct CertificateSource { ... }
  public struct PrivateKeySource { ... }
  public struct CertificateVerification { ... }
}

extension HTTP2ServerTransporti.Posix.Config { 
  public struct TLS {
    public var certificateChain: [TLSConfig.CertificateSource]
    // etc.
    
    public static func defaults(...) { ... }
    // etc.
  }
}
```

Reply:
Ah I see, sorry, I think it's clear now. Fixed a few docs too